packages/variable-resolver/src/browser/variable-resolver-service.ts

```diff
@@ -17,11 +17,12 @@
 // tslint:disable:no-any
 
 import { injectable, inject } from 'inversify';
-import { VariableRegistry } from './variable';
+import { Variable, VariableRegistry } from './variable';
 import URI from '@theia/core/lib/common/uri';
 
 export interface VariableResolveOptions {
     context?: URI;
+    overrides?: Variable[];
```

Comment:
It seems that a map of key->value will be enough for now, i.e. `values: { [key: string]: string }`